rdprail-shell/shell.c

```diff
@@ -1506,23 +1499,6 @@ handle_keyboard_focus(struct wl_listener *listener, void *data)
 		if (shsurf)
 			shell_surface_gain_keyboard_focus(shsurf);
 	}
-
-	if (new_focused_surface == shell->focus_proxy_surface) {
-		/* When new focused window is focus proxy window, client side window is

```

Comment:
we need this logic when RAIL window lost the focus. This is because RDP doesn't send key up event if focus is moved away while some key is still pressed. Since wayland compositor let wayland client (including xwayland) to repeat key input by itself, key remains pressed and repeats while focus is away to Wimdows's local window.

Reply:
What should happen is that the wayland compositor will send a "keyboard leave" event when the RAIL windows loses focus - this lets the client do some bookkeeping (like disarming its key repeat timer).

When the client gains focus again it receives a keyboard_enter event with a keymap of held keys.

Client's shouldn't repeat keys after they've lost keyboard focus, and the compositor shouldn't need to clear out their keymaps with key events before sending a leave.

Reply:
Can you see if https://github.com/microsoft/wslg/issues/207, especially https://github.com/microsoft/wslg/issues/207#issuecomment-842622670, doesn't occur with your patch applied?

Btw, we did fix the issue RDP client resending modifier key by changing RDP client which is mentioned at https://github.com/microsoft/wslg/issues/207#issuecomment-938757008.

Reply:
Ah, thanks for the pointer. I have caused a regression in this case.